### ISSUE TITLE
A11y_.NETCore3.0(WPF)_WPFConcepts_MenuSample_Screenreader: Screen reader is announcing wrong ordering for the file /edit / Project menuitem 

### ISSUE THREAD

**Issue by grubioe** [dotnet/wpf#1467]
Copied from VSO[ 956410](https://devdiv.visualstudio.com/DevDiv/_queries/edit/956410/)

Check out Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

Environment Details:

Application Name: .NET Core 3.0 WPF

.NET Core SDK version:  3.0.100-preview8-013442
Windows Version: Windows10

Screen Reader Vesion : Narrator, NVDA 2019.1.1

Pre-requisite:

1) Download the latest .NET Core 3 daily build by going to this link.
2) Install Windows (x86 or x64) versions of sdk listed under Master (3.0.x Runtime) by clicking "Installer" link.

Repro Steps:

1) Run exe file for "Concepts" from the build app folder provided. 
2) Main Window for "Concepts" solution will appear. 
3) Select "Samples" tab item. 
4) From right navigation under "Controls" select "Menu Sample". 
5) Navigate to the preview tab and the to "File"/"Edit" menu item.
6) start screen reader and observe the screen reader announcement.

Actual:

Screen reader is announcing wrong ordering for the file /edit / Project menu item.
for example there are four sub menu inside the Edit menu item but screen reader is announcing for first sub menu item as 1 of 5.
Expected:

Screen reader should announce proper ordering for sub menu item.

User Impact:

Screen reader user may get confuse with ordering inside the menuitem.

Recommendations:

Please refer to below URL which is repository for accessibility bug fixes.

https://microsoft.sharepoint.com/teams/msenable/Pages/Home.aspx

MAS Reference:

MAS1.3.1

https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

**Comment by rladuca** [comment]
Generally, this seems that a menu separator is being counted towards the number of menu items when it should not be.

Initial SizeOfSet work was done here: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/488213

There is possibly some missing code regarding separators in Menus.

**Comment by rladuca** [comment]
Setting estimate to 2 as I think this will just be a matter of figuring out how to filter out separators in the MenuItemAutomationPeer instead of directly using the ItemAutomationPeer.  Since the owning MenuItem should have knowledge of all internal pieces it can alter the count appropriately.

**Comment by arpitmathur** [comment]
Rob's assertion was correct. I was able to filter out separators in the MenuItemAutomationPeer itself.

The fix for SizeOfSet was adding a simple filter for separators, while the fix for PositionInSet was adding a filter for separators with a short-circuit when you're at the element whose position you're trying to find.

**Comment by arpitmathur** [comment]
Reopening the issue to track porting the change to release/3.1
